src/hardware/addon-protocols/rs232-usb.md

```diff
@@ -594,180 +594,415 @@ This is an example of binary data packets for image data transferred using the M
 
 ### Appendix D: Using protobuf Messages to Communicate by using 0x8C, 0x26
 
-Messages sent from the IOX to the GO is represented by IoxToGo
-Messages sent from the GO to the IOX is represented by IoxFromGo
-
-example of building protobuf in python:
-<code>
-iox_to_go=iox_messaging_pb2.IoxToGo()
-iox_to_go.pub_sub.sub.topic = 1 #TOPIC_ACCEL
-iox_to_go_seralized = iox_to_go.SerializeToString()
-</code>
-After the protobuf is built, we will get payload "0a040a020801"
-
-The content of complete .proto is as below.
-
-<code>
-// Possible subscription topics
-// Includes status data IDs
-enum Topic {
-    TOPIC_UNSPECIFIED = 0;
-    TOPIC_ACCEL = 1;
-    TOPIC_GPS = 2;
-    TOPIC_BATTERY_VOLTAGE = 3;
-    TOPIC_VIN = 4;
-    TOPIC_GEAR = 5;
-    TOPIC_ENGINE_SPEED = 6;
-    TOPIC_ENGINE_LOAD = 7;
-    TOPIC_ODOMETER = 8;
-    TOPIC_ACCEL_PEDAL_PERCENTAGE = 9;
-    TOPIC_COOLANT_TEMP = 10;
-    TOPIC_DOC_INTAKE_GAS_TEMP = 11;
-    TOPIC_DOC_OUTLET_GAS_TEMP = 12;
-    TOPIC_FUELTANK1_UNITS = 13;
-    TOPIC_FUELTANK2_UNITS = 14;
-    TOPIC_FUELTANK1_PERCENT = 15;
-    TOPIC_FUELTANK2_PERCENT = 16;
-    TOPIC_STATE_OF_CHARGE = 17;
-    TOPIC_ENGINE_ROAD_SPEED = 18;
-    TOPIC_VEHICLE_ACTIVE = 19;
-    TOPIC_DRIVER_SEATBELT = 20;
-    TOPIC_LEFT_TURN_SIGNAL = 21;
-    TOPIC_RIGHT_TURN_SIGNAL = 22;
-    TOPIC_EV_CHARGING_STATE = 23;
-    TOPIC_PARK_BRAKE = 24;
-}
+<a name="pubsub-proto"></a>
+<p align="right"><a href="#top">Top</a></p>
 
-message SubAck {
-    enum Result {
-        SUB_ACK_RESULT_UNSPECIFIED = 0;
-        // Subscription success
-        SUB_ACK_RESULT_SUCCESS = 1;
-        // Generic subscription failure
-        SUB_ACK_RESULT_FAILED = 2;
-        // Subscribe fails if an unknown topic is specified
-        SUB_ACK_RESULT_UNKNOWN_TOPIC = 3;
-        // Subscribe fails if the topic has already been subscribed to
-        SUB_ACK_RESULT_TOPIC_ALREADY_SUBBED = 4;
-        // Unsubscribe fails if the topic has not been subscribed to
-        SUB_ACK_RESULT_TOPIC_NOT_SUBBED = 5;
-        // Unsubscribe fails if the subscription belongs to another IOX.
-        SUB_ACK_RESULT_SUBSCRIPTION_NOT_AVAILABLE = 6;
-    }
-    Result result = 1;
-    Topic topic = 2;
-}
+## pubsub.proto
```

Comment:
In new MD, I replaced this wrong name.

Reply:
It is now `#### PUB/Sub Protocol Documentation`.